app/graphql/resolvers/base_resolver.rb

```diff
@@ -37,7 +37,9 @@ def admin?
   end
 
   def matching_user(submission, session_id)
-    submission.user&.gravity_user_id == @context&.[](:current_user) ||
-      submission.session_id == session_id
+    (
+      submission.user&.gravity_user_id.present? &&
+        submission.user&.gravity_user_id == @context&.[](:current_user)
+    ) || (session_id.present? && submission.session_id == session_id)
```

Comment:
**suggestion:** this conditional is becoming difficult to read - could you refactor this so that the condition is easier to understand? (ie. `submitted_by_current_user? || submitted_by_current_session?)